test/MathUtils.test.cpp

```diff
@@ -0,0 +1,237 @@
+#include "NAS2D/MathUtils.h"
+#include "NAS2D/Renderer/Primitives.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+TEST(MathUtils, divideUp)
+{
+	EXPECT_DEATH(NAS2D::divideUp(0, 0), "divideUp(0,0) divide by zero.");
```

Comment:
This looks like it could be an [`EXPECT_THROW`](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#exception-assertions) rather than an `EXPECT_DEATH`.

I'm wondering if the divide by 0 cases should be separated into their own test case, or perhaps segregated from the other test lines. I think I would have done that due to the different behavior in that case.

Reply:
I just realized, this is technically incorrect. For integers, division by 0 is undefined. We can't even rely on the code dying. It's just extremely common for the OS to throw a system level exception that causes the code to die.

We should probably update the code to detect this case and throw an exception, and then update the test code to check for an exception. That way we would have defined behaviour.